### ISSUE TITLE
[Bug]: Netdata cannot compile on FreeBSD

### ISSUE THREAD

**Issue by thiagoftsm** [netdata/netdata#12793]
### Bug description

After we merge the PR https://github.com/netdata/netdata/pull/12765 when we try to compile `netdata` on  `FreeBSD 13.0-RELEASE-p11` we receive the following error:

```sh
libnetdata/libnetdata.c: In function 'madvise_dontdump':
libnetdata/libnetdata.c:970:33: error: 'MADV_DONTDUMP' undeclared (first use in this function); did you mean 'MADV_DONTFORK'?
  970 |     int ret = madvise(mem, len, MADV_DONTDUMP);
      |                                 ^~~~~~~~~~~~~
      |                                 MADV_DONTFORK

libnetdata/libnetdata.c:970:33: note: each undeclared identifier is reported only once for each function it appears in
libnetdata/libnetdata.c: In function 'madvise_mergeable':
libnetdata/libnetdata.c:975:43: warning: unused parameter 'mem' [-Wunused-parameter]
  975 | static inline int madvise_mergeable(void *mem, size_t len) {
      |                                     ~~~~~~^~~
libnetdata/libnetdata.c:975:55: warning: unused parameter 'len' [-Wunused-parameter]
  975 | static inline int madvise_mergeable(void *mem, size_t len) {
      |                                                ~~~~~~~^~~
gmake[2]: *** [Makefile:5618: libnetdata/libnetdata.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [Makefile:6325: all-recursive] Error 1
gmake: *** [Makefile:3458: all] Error 2
 FAILED  ''

 ABORTED  Failed to build Netdata.
```

### Expected behavior

As user I want to compile netdata on FreeBSD

### Steps to reproduce

1. Clone current netdata master branch
2. Try to compile it.

### Installation method

from source

### System info

```shell
FreeBSD freebsd 13.0-RELEASE-p11 FreeBSD 13.0-RELEASE-p11 #0: Tue Apr  5 18:54:35 UTC 2022     root@amd64-builder.daemonology.net:/usr/obj/usr/src/amd64.amd64/sys/GENERIC  amd64
```

### Netdata build info

```shell
I cannot fill this information, because I cannot compile
```

### Additional info

I also observed warnings for `apps.plugin`, but  prefer to open another issue when I have conditions to compile netdata and run to give a complete report.
